File: solutions/_1436_destination_city/solution.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
var destCity = function (paths) {
  const cities = new Set();
  for (const path of paths) {
    cities.add(path.at(0));
  }
  for (const path of paths) {
    if (!cities.has(path.at(1))) {
      return path.at(1);
    }
  }
};

module.exports = { destCity };

/*
https://leetcode.com/problems/destination-city/
*/

File: solutions/_1436_destination_city/solution.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

class Solution
{
public:
    string destCity(vector<vector<string>> &paths)
    {
        set<string> cities;
        for (auto path : paths)
        {
            cities.insert(path[0]);
        }
        for (auto path : paths)
        {
            if (cities.find(path[1]) == cities.end())
            {
                return path[1];
            }
        }
        return "";
    }
};

/*
https://leetcode.com/problems/destination-city/
*/